### PR TITLE
Feat/docker builds

### PR DESCRIPTION
**Pull request by clementnuss**
I've added simple docker builds using [`ko-build/ko`](https://github.com/ko-build/ko), I think it would quite useful to have docker images available.

let me know if you want me to adapt the README.

also, if this gets merged, I'll create another PR with a Kubernetes `CronJob` using the docker image to defragment my etcd clusters.

thanks for the tool by the way! really useful :)

### CODE REVIEW

**Comment by ahrtr** [comment]
Thank you for the PR. This tool doesn't have an official image registry for now. [github registry](https://github.com/features/packages#pricing) has some limitation, might not be a good default choice. 

So I'd like to leave users to build & push the image themselves. But of course, if you want to improve the existing [release script](https://github.com/ahrtr/etcd-defrag/blob/main/release.sh) to include the functionality of building & pushing image, then I am happy to accept it. But the functionality should can be enabled/disabled by flags (e.g. --build-image, --push-image), and should be disabled by default.

**Comment by clementnuss** [comment]
I see your point. I think would go another way however, for simplicity reasons: I would fork that repo and release a docker container under `postfinance/etcd-defrag` if that's ok for you. I would then update my fork based on yours from times to times.

This makes it quite easy to keep up-to-date containers.
modifying the build script comes with the challenge that I'd need to export an access token and to build locally, something I don't want to do.

**Comment by clementnuss** [comment]
or I could also modify the gh-action in the same way as I did in this other project I'm maintaining: https://github.com/postfinance/kubelet-csr-approver/blob/main/.github/workflows/publish.yaml#L129

that way there would be container images published to ghcr.io and docker.io. both come with some limitations, but with two approaches I would find it acceptable.

also, having the container named `ahrtr/etcd-defrag` would probably increase the adoption of this tool, it then makes it really simple to create a Kubernetes CronJob and to defragment in-cluster

**Comment by ahrtr** [comment]
Re-read the https://github.com/features/packages#pricing, it seems that the 1GB limitation of "`Data transfer out outside of Actions`" is only for private repositories, and public repositories (such as ahrtr/etcd-defrag) do not have such limitation. If it's true, it might be OK to use github package registry.

File: .github/workflows/etcd-defrag.yaml
Comment (ahrtr) [inline .github/workflows/etcd-defrag.yaml:18]: Please reuse the golang version defined in [go-version.yaml](https://github.com/ahrtr/etcd-defrag/blob/main/.github/workflows/go-version.yaml), please refer to [test.yaml](https://github.com/ahrtr/etcd-defrag/blob/main/.github/workflows/test.yaml).

File: .github/workflows/etcd-defrag.yaml
Comment (ahrtr) [inline .github/workflows/etcd-defrag.yaml:12]: If I understand it correctly, the workflow is triggered on each push event, but it will not execute if the latest commit isn't tagged. Correct?

File: .github/workflows/etcd-defrag.yaml
Comment (ahrtr) [inline .github/workflows/etcd-defrag.yaml:30]: What's the motivation to release untagged image?

File: .github/workflows/etcd-defrag.yaml
Comment (clementnuss) [reply .github/workflows/etcd-defrag.yaml:12]: correct, I use that to differentiate between development builds (nightlies) and tagged release.

File: .github/workflows/etcd-defrag.yaml
Comment (clementnuss) [reply .github/workflows/etcd-defrag.yaml:30]: when developping a new feature for example, I like the ability to test it directly through using the `etcd-defrag:dev` tag. could also be named `:nightly`.

this can be removed if you don't see the usage for this

Comment (ahrtr) [reply .github/workflows/etcd-defrag.yaml:30]: I suggest to remove it for now. I do not think there will be frequent changes or features for this repository.

Comment (clementnuss) [reply .github/workflows/etcd-defrag.yaml]: good point. done

File: .github/workflows/etcd-defrag.yaml
Comment (ahrtr) [inline .github/workflows/etcd-defrag.yaml:30]: The line 29 has flag `--tags $TAG`, but line 30 doesn't have. Does it mean that line 30 generate an image with tag "`latest`"?

Comment (clementnuss) [reply .github/workflows/etcd-defrag.yaml:30]: exactly

Comment (ahrtr) [reply .github/workflows/etcd-defrag.yaml:30]: thx

**Comment by ahrtr** [comment]
Could you please also update https://github.com/ahrtr/etcd-defrag#container-image?

**Comment by ahrtr** [comment]
> Could you please also update https://github.com/ahrtr/etcd-defrag#container-image?

Never mind, we can take care of it in a separate PR. I can also take care of it.

**Comment by ahrtr** [comment]
I will create a new tag & publish a new release after merging this PR.

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you! @clementnuss

**Pull request merged**
